src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -32,41 +39,71 @@ const CreateCourse = ({ authorsList }) => {
 		setAvailableAuthors((prevState) => [author, ...prevState]);
 	};
 
+	const onAddTitleHandler = (title) => setTitle(title);
+	const onAddDescriptionHandler = (description) => setDescription(description);
+	const onAddDurationHandler = (duration) => setDuration(duration);
+	const onCreateCourseHandler = (event) => {
+		event.preventDefault();
+		const newCourse = {
+			id: Math.random(),
+			title,
+			description,
+			creationDate: new Date(),
+			duration,
+			authors: courseAuthors,
+		};
+
+		//Uladzislau, I need your advice on what's the right way to pass this new course to /courses:
+		// I was not able to come up with a solution on how to pass this new course object to AND save all the courses added previously
+		// the options I tried (and they did not work properly, the courses added previously were lost):
+		// 1) pass as a state in navigate()
+		// 2) save in localStorage
+		// 3) lift it up by passing through props
```

Comment:
You should put `mockedCoursesList` and `mockedAuthorsList` into a local state in `App` component and pass this state and state updaters to your components.

```suggestion
// in App.jsx
const [courses, setCourses] = React.useState(mockedCoursesList);
const [authors, setAuthors] = React.useState(mockedAuthorsList);

<Route
	path='/courses'
	element={
		<Courses
			coursesList={courses}
			authorsList={authors}
			setAuthors={setAuthors}
		/>
	}
/>
<Route
	path='/courses/add'
	element={(
		<CreateCourse
			authorsList={authors}
			setAuthors={setAuthors}
			onCourseAdd={(newCourse) => {
				setCourses(prevCourses => [...prevCourses, newCourse])
			}}
		/>
	)}
/>


// in CreateCourse.jsx

const onCreateCourseHandler = (event) => {
	event.preventDefault();
	const newCourse = {
		id: Math.random(),
		title,
		description,
		creationDate: new Date(),
		duration,
		authors: courseAuthors,
	};
	
	onCourseAdd(newCourse)
	navigate('/courses');
```